internal/clone/registry.go

```diff
@@ -0,0 +1,141 @@
+package clone
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"fmt"
+
+	"github.com/Microsoft/hcsshim/internal/regstate"
+	"github.com/Microsoft/hcsshim/internal/uvm"
+)
+
+const (
+	configRoot = "LateClone"
+	configKey  = "UVMConfig"
+)
+
+// When encoding interfaces gob requires us to register the struct types that we will be
+// using under those interfaces. This registration needs to happen on both sides i.e the
+// side which encodes the data (i.e the shim process of the template) and the side which
+// decodes the data (i.e the shim process of the clone).
+// Go init function: https://golang.org/doc/effective_go.html#init
+func init() {
+	// Register the pointer to structs because that is what is being stored.
+	gob.Register(&uvm.VSMBShare{})
```

Comment:
Probably would be better to move these registrations to a `Once` called from where they are used, but that could be a future optimization.

Reply:
Do you prefer using `Once` over this? Is there any specific reason?

Reply:
`init` will be called if this package is referenced, even if we never actually use the clone functionality. Using a `Once` allows us to defer the initialization work until it's actually needed, while still ensuring we only do it one time.

Reply:
That makes sense. I think we can do it later as a future optimization if that's okay.

Reply:
Yes, that's fine.